assets/admin/editor-wizard/steps/course-details-step.js

```diff
@@ -16,71 +20,79 @@ import LimitedTextControl from '../../../blocks/editor-components/limited-text-c
-	// TODO Replace this sample implementation.
-	const updateNewCourseTitle = ( value ) => {
-		setWizardData( { ...wizardData, newCourseTitle: value } );
+	usePostTitle( wizardData.courseTitle );
+
+	const updateCourseTitle = ( value ) => {
+		setWizardData( { ...wizardData, courseTitle: value } );
 	};
-	const updateNewCourseDescription = ( value ) => {
+
+	const updateCourseDescription = ( value ) => {
 		setWizardData( {
 			...wizardData,
-			newCourseDescription: value,
+			courseDescription: value,
 		} );
 	};
+
 	return (
 		<div className="sensei-editor-wizard-modal__columns">
 			<div className="sensei-editor-wizard-modal__content">
 				<h1>Course Details Step</h1>
-				<div>
+				<div className={ 'sensei-editor-wizard-step__description' }>
+					{ __(
+						'Keep your Course Title short as it will get displayed in different places around your website. You can easily change both later.',
+						'sensei-lms'
+					) }
+				</div>
+				<div className={ 'sensei-editor-wizard-step__form' }>
 					<LimitedTextControl
 						label={ __( 'Course Title', 'sensei-lms' ) }
-						value={ wizardData.newCourseTitle ?? '' }
-						onChange={ updateNewCourseTitle }
+						value={ wizardData.courseTitle ?? '' }
+						onChange={ updateCourseTitle }
 						maxLength={ 40 }
 					/>
 					<LimitedTextControl
 						label={ __( 'Course Description', 'sensei-lms' ) }
-						value={ wizardData.newCourseDescription ?? '' }
-						onChange={ updateNewCourseDescription }
+						value={ wizardData.courseDescription ?? '' }
+						onChange={ updateCourseDescription }
 						maxLength={ 350 }
 						multiline={ true }
 					/>
 				</div>
-				<div>PENDING TO IMPLEMENT</div>
 			</div>
 			<div className="sensei-editor-wizard-modal__illustration">
 				<img
 					src={
 						window.sensei.pluginUrl +
-						'assets/dist/images/sensei-pro-upsell.png'
+						'assets/dist/images/course-details-step.png'
 					}
-					alt="PENDING TO IMPLEMENT, BUT HERE TO SHOW IT WORKING"
+					alt={ __(
+						'Illustration of course sample with some placeholders.',
+						'sensei-lms'
+					) }
 					className="sensei-editor-wizard-modal__illustration-image"
 				/>
 			</div>
 		</div>
 	);
 };
 
-CourseDetailsStep.Actions = ( { data, goToNextStep } ) => {
-	// Actions have access to the whole wizard data.
-	const secondaryAction = () => {
-		// TODO Remove this.
-		// eslint-disable-next-line no-alert
-		window.alert( `Data ${ JSON.stringify( data ) }` );
-	};
+CourseDetailsStep.Actions = ( { goToNextStep } ) => {
 	return (
 		<div>
-			<button onClick={ secondaryAction }>
-				<span
-					role="img"
-					aria-label="Funny eyes that will be removed later."
-				>
-					👀
-				</span>
-			</button>
-			<button onClick={ goToNextStep }>Next</button>
+			<Button isPrimary onClick={ goToNextStep } target="_blank">
+				{ __( 'Continue', 'sensei-lms' ) }
+			</Button>
 		</div>
 	);
 };
 
+function usePostTitle( title ) {
```

Comment:
Hey, not sure if you're gonna follow @renatho's suggestion [here](https://github.com/Automattic/sensei/pull/5185/files#r881943534), but if not, I'd like to remind you to prefer to use something like this here:

```suggestion
const usePostTitle = ( title ) => {
```

Just to keep the PR aligned with the standards of the project. =)

Reply:
Finally did as Renatho suggested, but thanks for pointing this out!